src/commands/mdn.ts

```diff
@@ -4,13 +4,22 @@ import type { Command } from "djs-fsrouter";
 import scrape, { elementToMarkdown } from "../scraper.ts";
 import type { Element } from "cheerio";
 
-const MDN_URL = "https://developer.mozilla.org/en-US/";
-type SearchResult = {
+import { truncate } from "../utils/common.ts";
+
+import { Index } from "flexsearch";
+
+const MDN_ROOT = "https://developer.mozilla.org";
+const MDN_INDEX = `${MDN_ROOT}/en-US/search-index.json`;
+const DOCS_ROOT = "/en-US/docs/";
+const RESULT_INDICATOR = "d#:";
+type IndexEntry = {
 	title: string;
-	link: string;
-	snippet: string;
+	url: string;
 };
 
+let index: IndexEntry[];
+const searcher = new Index({ tokenize: "forward" });
```

Comment:
Have you tested out some other encoders? I feel like it would be helpful to at least normalize case or perhaps even use soundex transformations (although idk how well they would work for technical terms). At the very least, this should improve compression of the index a bit.

Also, have you benchmarked FlexSearch's cache yet? It can be pretty helpful for query performance.

Reply:
I have. Strict tokenizer doesn't find results if the words are in the wrong order or incomplete.
reverse and full tokernizers don't seem to do much for our use case and significantly increase the cache size so I didn't use them.
Not sure how useful soundex would be but I can try it.

Reply:
I was more referring to [encoders](https://github.com/nextapps-de/flexsearch?tab=readme-ov-file#encoders), not tokenizers.

Reply:
Aye, the "advanced" encoder saves about 10 MB of memory and handles typos slightly better. The way to import it is a bit ugly